Common/src/main/java/com/blamejared/crafttweaker/api/tag/MCTag.java

```diff
@@ -3,231 +3,121 @@
 import com.blamejared.crafttweaker.api.annotation.ZenRegister;
 import com.blamejared.crafttweaker.api.bracket.CommandStringDisplayable;
 import com.blamejared.crafttweaker.api.tag.manager.ITagManager;
-import com.blamejared.crafttweaker.api.util.Many;
+import com.blamejared.crafttweaker.api.tag.type.KnownTag;
+import com.blamejared.crafttweaker.api.tag.type.UnknownTag;
 import com.blamejared.crafttweaker_annotations.annotations.Document;
+import net.minecraft.core.Holder;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.tags.Tag;
+import net.minecraft.tags.TagKey;
 import org.openzen.zencode.java.ZenCodeType;
 
 import javax.annotation.Nonnull;
-import java.util.Arrays;
-import java.util.Iterator;
 import java.util.List;
-import java.util.stream.Collectors;
 
 /**
- * A reference to a Tag object.
- * Note that this tag may not exist in the game already, such as when you create new tags.
- * See the {@link MCTag#exists()} Method on whether this tag already exists.
- * <p>
- * A tag will be created as soon as you add
- *
- * @docParam this <tag:items:forge:gems>
+ * @docParam this <tag:items:minecraft:wool>
  */
 @ZenRegister
 @Document("vanilla/api/tag/MCTag")
 @ZenCodeType.Name("crafttweaker.api.tag.MCTag")
-public final class MCTag<T> implements CommandStringDisplayable, Iterable<T> {
-    
-    private final ResourceLocation id;
-    private final ITagManager<T> manager;
-    
-    public MCTag(ResourceLocation id, ITagManager<T> manager) {
-        
-        this.id = id;
-        this.manager = manager;
-    }
+public interface MCTag extends CommandStringDisplayable, Comparable<MCTag> {
```

Comment:
Unknown Tag does need to be a separate from MCTag because there are different implementations based on different usages, for example, 
```java
UnknownTag implements MCTag, Iterable<ResourceLocation>
```
but
```java
KnownTag<T> implements MCTag, Iterable<T>
```

Wouldn't be possible otherwise.

I also want to make KnownTag auto cast to UnknownTag just for when a mod asks for an UnknownTag for say Menu, if we made Menu known, that mod would have issues